includes/multi-currency/MultiCurrency.php

```diff
@@ -181,20 +178,29 @@ class MultiCurrency {
 	 */
 	private $order_meta_helper;
 
+	/**
+	 * Gateway context.
+	 *
+	 * @var array
```

Comment:
Should we have a "gateway context" interface to document what can or cannot be passed to this variable? That would make things clear to whatever gateway integrating with multi-currency.

Reply:
The array indeed looks weird, and that's because these variables don't fit anywhere in the module. 😞 

I have made a few improvements:

- fa0aa8386c7f562abfd75d05ef08051e2bc89ab5: Removed `is_customer_multi_currency_enabled` from the context since I noticed we have [this check](https://github.com/Automattic/woocommerce-payments/blob/b1b0e0d1a9c0b7061e2ce2bba28df155427572bc/includes/multi-currency/wc-payments-multi-currency.php#L25) and if `WC_Payments_Features::is_customer_multi_currency_enabled()` is `false` multi-currency shouldn't even be available. Using this in the module apparently was a leftover from a previous iteration.
- d46fa82f05a99dcbd33092991ab975a413ab0844: Similar to `class_exists( 'WC_Subscriptions' )` we can do `class_exists( 'WC_Payments_Subscriptions' )` to check if WCPay subscriptions is enabled. That appears to be the right approach actually, since there are some required checks [for Stripe billing as well](https://github.com/Automattic/woocommerce-payments/blob/b1b0e0d1a9c0b7061e2ce2bba28df155427572bc/includes/class-wc-payments.php#L694-L697).
- b1b0e0d1a9c0b7061e2ce2bba28df155427572bc: Introduced a `get_context_for_multi_currency` method to return `is_dev_mode` and plugin constants. I guess that should be enough for an intermediate solution instead of an interface where we don't really need to implement any methods.

Ideally several things should be refactored, but given time constraints we have to be ok with some code quality issues.

Reply:
Thanks for doing these improvements 👍 They look good and make sense to me.